lib/web/integrations.go

```diff
@@ -219,6 +223,79 @@ func (h *Handler) integrationsDelete(w http.ResponseWriter, r *http.Request, p h
 	return OK(), nil
 }
 
+// integrationsDelete removes an Integration based on its name
+func (h *Handler) integrationsDeleteWithCleanup(w http.ResponseWriter, r *http.Request, p httprouter.Params, sctx *SessionContext, site reversetunnelclient.RemoteSite) (interface{}, error) {
+	values := r.URL.Query()
+	orgName := values.Get("orgName")
+
+	integrationName := values.Get("name")
+	if integrationName == "" {
+		return nil, trace.BadParameter("an integration name is required")
+	}
+
+	integrationKind := values.Get("subKind")
+	if integrationKind == "" {
+		return nil, trace.BadParameter("the integrations subKind is required")
+	}
+
+	// Validate subKind support.
+	switch integrationKind {
+	case types.KindGithub:
+		if orgName == "" {
+			return nil, trace.BadParameter("the integrations orgName is required")
+		}
+	default:
+		return nil, trace.BadParameter("deleting integration with clean up for subKind %q not supported", integrationKind)
+	}
+
+	clt, err := sctx.GetUserClient(r.Context(), site)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if err := clt.DeleteIntegration(r.Context(), integrationName); err != nil {
+		if !trace.IsNotFound(err) {
+			return nil, trace.Wrap(err)
+		}
+		// Ignore not found errors. It could have already been deleted if a user
+		// is retrying from failure to delete git servers for whatever reason.
+	}
```

Comment:
```suggestion
	if err := clt.DeleteIntegration(r.Context(), integrationName); err != nil {
		// Ignore not found errors. It could have already been deleted if a user
		// is retrying from failure to delete git servers for whatever reason.
		if !trace.IsNotFound(err) {
			return nil, trace.Wrap(err)
		}
	}
```

- Move the comment closer to the code it describes.
- We're still mentioning "delete git servers" even though this is a generic func.